redis.go

```diff
@@ -146,16 +144,14 @@ func blocking(
 	}
 }
 
-func setCondTimer(sig *sync.Cond, timedOut *bool, timeout time.Duration, cleanup chan struct{}) {
-	dl := time.NewTimer(timeout)
-	defer dl.Stop()
+func setCondTimer(ctx context.Context, sig *sync.Cond, timedOut *bool, timeout time.Duration) {
 	select {
-	case <-dl.C:
+	case <-time.After(timeout):
```

Comment:
Using time.After is totally fine. I used the NewTimer/Stop construct because of this documentation to time.After:
> The underlying Timer is not recovered by the garbage collector until the timer fires. If efficiency is a concern, use NewTimer instead and call Timer.Stop if the timer is no longer needed.

Reply:
Ah, okay. I'll take that out then.